source/tinyengine.c

```diff
@@ -0,0 +1,52 @@
+#include "tinyengine_platform.h"
+
+static void
+Tiny_ProcessDigitalButton(tiny_digital_button *Button, b32 WasDown)
+{
+    // TODO(hayden): Avoid wrapping Up/Down?
+    Button->Down = WasDown ? ++Button->Down : false;
+    Button->Up = !WasDown ? ++Button->Up : false;
+    Button->Pressed = (Button->Down == 1);
+    Button->Released = (Button->Up == 1);
+}
+
+b32 KeyboardState[TINY_EVENT_MAX_KEYS]; // NOTE: true == IsDown
+tiny_digital_button Keyboard[TINY_EVENT_MAX_KEYS];
+
+static void
+Tiny_Update(tiny_platform *Platform)
+{
+    // TODO(hayden): Normalize mouse input here?
+
+    { // Update input events
+        // Update KeyboardState with new events
+        for(int EventIndex = 0; EventIndex < Platform->EventQueueIndex; ++EventIndex)
```

Comment:
while Tiny_GetMessage maybe? to iterate through all of this instead of hard coding it.